### PR TITLE
Rewrok applications examples and server tests

### PR DESCRIPTION
**Pull request by artemklevtsov**
All examples moves to `inst/examples`.

Added:
- `rserve`: app as reference
- `hello`: app with query and path template handling
- `fibonacci-numbers`: app with query param validation
- `static`: app with handle static file and static directory
- `auth-basic`: app with basic http auth handling
- `auth-bearer`: app with bearer http auth handling
- `plot-*`: app with plot
- `middleware`: app with middlewares
- `openapi`: app with OpenAPI spec and Swagger UI

Remove tests with running server. Added tests with `process_request` method instead.

### CODE REVIEW

File: inst/examples/html-form/app.R [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+## ---- load packages ----
+
+library(RestRserve)
+library(htmltools)
+
+
+## ---- create handler for the HTTP requests ----
+
+# simple response
+form_handler = function(request, response) {
```
Comment (dselivanov) [inline inst/examples/html-form/app.R]: this is very cool 👍

File: inst/examples/middleware/app.R [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+stop_handler = function(request, response) {
+  stop("unexpected")
+}
+
+
+## ---- create middleware ----
+
+mw1 = RestRserveMiddleware$new(
+  process_request = function(req, res) {
```
Comment (dselivanov) [inline inst/examples/middleware/app.R]: let us use `request` and `response` names everywhere

File: inst/examples/openapi/app.R [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+app$add_swagger_ui(
+  path = "/swagger",
+  path_openapi = "/openapi.yaml",
+  path_swagger_assets = "/swagger/assets/",
+  file_path = tempfile(fileext = ".html"),
+  use_cdn = FALSE
+)
+
+app$add_static(
+  path = "/redoc",
```
Comment (dselivanov) [inline inst/examples/openapi/app.R]: 👍cool, didn't know about redoc

File: inst/examples/plot-base/app.R [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  response$content_type = "image/png"
+  response$status_code = 200L
+  response$serializer = identity
+}
+
+
+## ---- create application -----
+
+app = RestRserveApplication$new(
+  content_type = "text/plain",
```
Comment (dselivanov) [inline inst/examples/plot-base/app.R]: we should omit `text/plain` as it is confusing for serving png. Or provide `image/png` to the app

File: inst/examples/plot-ggplot/app.R [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  response$content_type = "image/png"
+  response$status_code = 200L
+  response$serializer = identity
+}
+
+
+## ---- create application -----
+
+app = RestRserveApplication$new(
+  content_type = "text/plain",
```
Comment (dselivanov) [inline inst/examples/plot-ggplot/app.R]: "image/png"

File: inst/examples/plot-magick/app.R [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  response$content_type = mimemap[[img_type]]
+  response$status_code = 200L
+  response$serializer = identity
+}
+
+
+## ---- create application -----
+
+app = RestRserveApplication$new(
+  content_type = "text/plain",
```
Comment (dselivanov) [inline inst/examples/plot-magick/app.R]: "image/png"

File: inst/examples/predict-lm/app.R [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+set.seed(0)
+x = rnorm(100)
+y = x + rnorm(100)
+model = lm(y ~ x)
+rm(x, y)
+
+
+## ---- create serializer ----
+
+to_json = function(x) {
```
Comment (dselivanov) [inline inst/examples/predict-lm/app.R]: how about RestRserve::to_json?

File: inst/examples/auth-basic/app.R [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+app$add_get(
+  path = "/securearea/{resource}",
+  FUN = securearea_handler,
+  match = "regex"
+)
+
+
+## ---- start application ----
+
+if (isTRUE(mget("run_app", ifnotfound = TRUE)$run_app)) {
```
[the remark this answers is not in the document]
Comment (artemklevtsov) [reply inst/examples/auth-basic/app.R]: To control startup app in the tests. May be remove app$run() from the script or comment it.

File: inst/examples/compression/app.R [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+library(RestRserve)
+
+
+## ---- create handler for the HTTP requests ----
+
+# simple response
+hello_handler = function(request, response) {
+  resp_body = "Hello, World!"
+  enc = request$headers[["accept-encoding"]]
```
[the remark this answers is not in the document]
Comment (artemklevtsov) [reply inst/examples/compression/app.R]: Sure. When I wrote it this method does not exists yet :)

File: inst/examples/openapi/app.R (hunk 3 above)
Comment (artemklevtsov) [reply inst/examples/openapi/app.R]: imho, rapidoc better.

**Pull request merged**